AtlasUtil/include/AtlasUtil/Annotate.h

```diff
@@ -10,12 +11,69 @@ inline void SetBlockID(llvm::BasicBlock *BB, int64_t i)
     BB->getFirstInsertionPt()->setMetadata("BlockID", idNode);
 }
 
-inline void SetValueID(llvm::Value *val, int64_t i)
+inline void SetValueIDs(llvm::Value *val, int64_t &i)
```

Comment:
It will not overwrite prior IDs. Before each metadata node is generated, there's a check on each instruction that looks for a "ValueID" metadata node (because instructions can have both "ValueID" and "BlockID" metadata, unlike GVs). For GVs, it simply checks for any metadata. 
There are two cases to the recursion:
1.) There is no metadata. In that case we haven't seen this value or its operands yet. Set value metadata and recurse upon operands.
2.) There is metadata. That means we've seen this value's operands as well. base case for recursion.